accord-core/src/main/java/accord/local/Node.java

```diff
@@ -475,10 +477,15 @@ public void send(Id to, Request send)
         messageSink.send(to, send);
     }
 
-    public void reply(Id replyingToNode, ReplyContext replyContext, Reply send)
+    public void reply(Id replyingToNode, ReplyContext replyContext, Reply send, Throwable failure)
     {
-        // TODO (usability, now): add Throwable as an argument so the error check is here, every single message gets this wrong causing a NPE here
-        if (send == null)
+        if (failure != null)
+        {
+            checkState(send == null, "fail (%s) and send (%s) are both not null", failure, send);
```

Comment:
```suggestion
            if (send != null)
                agent().onUncaughtException(new IllegalArgumentException(String.format("fail (%s) and send (%s) are both not null", failure, send)));
```

this situation would currently cause us to loose the real exception